pages/search/index.js

```diff
@@ -1,48 +1,285 @@
 import styled from "styled-components";
 import VerticalGameList from "@/components/VerticalGameList";
-
+import Fuse from "fuse.js";
 import { useState } from "react";
 import useSWR from "swr";
+import { FaTimes } from "react-icons/fa";
+
+const INITIAL_FILTERS = {
+  name: "",
+  categories: "",
+  rating: "",
+  minAge: "",
+  players: "",
+  yearpublished: "",
+  playtime: "",
+};
 
-export default function SearchPage({ isFavorite, toggleFavorite, showModal }) {
+export default function SearchPage({ isFavorite, toggleFavorite }) {
   const { data } = useSWR("./api/games");
-  const [searchInput, setSearchInput] = useState("");
+  const [searchQuery, setSearchQuery] = useState("");
+
+  const [showFilters, setShowFilters] = useState(false);
+
+  const [filters, setFilters] = useState(INITIAL_FILTERS);
+
+  function fuzzySearch(data, query, threshold, keys) {
+    const fuseOptions = {
+      threshold: threshold,
+      keys: keys,
+    };
+
+    const fuse = new Fuse(data, fuseOptions);
+
+    if (query.length > 0) {
+      const fuzzy = fuse.search(query);
+      const fuzzyResults = fuzzy.map((object) => object.item);
+
+      return fuzzyResults;
+    } else {
+      return [];
+    }
+  }
+
+  function filterGames(data) {
+    const filteredResultsName = filters.name
+      ? fuzzySearch(data, filters.name, 0.6, ["name"])
+      : data;
+
+    const filteredResultsCategory = filters.categories
+      ? fuzzySearch(filteredResultsName, filters.categories, 0.1, [
+          "categories",
+        ])
+      : filteredResultsName;
 
-  const searchedGames =
-    searchInput.length > 0
-      ? data.filter((game) =>
-          `${game.name}`.toLowerCase().includes(searchInput.toLowerCase())
+    const filteredResultsRating = filters.rating
+      ? filteredResultsCategory.filter(
+          (game) => +game.rating >= +filters.rating
         )
-      : [];
+      : filteredResultsCategory;
+
+    const filteredResultsMinAge = filters.minAge
+      ? filteredResultsRating.filter((game) => +game.minAge >= +filters.minAge)
+      : filteredResultsRating;
+
+    const filteredResultsPlayers = filters.players
+      ? filteredResultsMinAge.filter(
+          (game) =>
+            +game.minPlayers <= +filters.players &&
+            +game.maxPlayers >= +filters.players
+        )
+      : filteredResultsMinAge;
+
+    const filteredResultsPlaytime = filters.playtime
+      ? filteredResultsPlayers.filter(
+          (game) =>
+            +game.minPlaytime <= +filters.playtime &&
+            +game.maxPlaytime >= +filters.playtime
+        )
+      : filteredResultsPlayers;
+
+    const filteredResultsYear = filters.yearpublished
+      ? filteredResultsPlaytime.filter(
+          (game) => +game.yearpublished === +filters.yearpublished
+        )
+      : filteredResultsPlaytime;
+
+    return filteredResultsYear;
+  }
+
+  function handleSubmit(event) {
```

Comment:
In my opinion it is a bit confusing to search while typing and then have a submit button to again start searching.  A better solution could be to render a message like 'no results' when the filteredData is empty.